src/fullsignalk.js

```diff
@@ -76,66 +76,114 @@ FullSignalK.prototype.deleteContext = function(contextKey) {
+ * vessels.foo.  Signalk tracks multiple vessels, each with their own context.
+ * This method returns the context for the vessel.  It additionally adds either
+ * the mmsi or url to the context to allow for easier access.
+ *
+ * For example:
+ * contextPath=vessels.urn:mrn:imo:mmsi:276810000
+ * before context={}
+ * after context={"mmsi":"276810000"}
+ *
+ * @param {Object} root the signalk store
+ * @param {string} contextPath the path to the desired vessel
+ * @return {Object} the context for the desired vessel from the signalk store
+ */
+function findContextAndUpdateIdentity(root, contextPath) {
+  // get the context and create an empty context if it doesn't exist
+  let context = _.get(root, contextPath);
   if(!context) {
     context = {};
     _.set(root, contextPath, context);
   }
-  var identity = contextPath.split('.')[1];
+
+  // contextPath is something like "vessels.foo" or "vessels.urn:mrn:..."
+  // if we have a full context path, add its contents to the context, so that
+  // we can more easily access the mmsi or url
+  const identity = contextPath.split('.')[1];
   if(!identity) {
     return undefined;
   }
   signalkSchema.fillIdentityField(context, identity);
+
   return context;
 }
 
 FullSignalK.prototype.addUpdates = function(context, contextPath, updates) {
-  var len = updates.length;
-  for (var i = 0; i < len; ++i) {
-    this.addUpdate(context, contextPath, updates[i]);
-  }
+  let that = this;
+  updates.forEach(function(update) {
+    that.addUpdate(context, contextPath, update);
+  });
 }
 
 FullSignalK.prototype.addUpdate = function(context, contextPath, update) {
+  // first, update the sources in the full context
   if (typeof update.source != 'undefined') {
     this.updateSource(context, update.source, update.timestamp);
   } else if(typeof update['$source'] != 'undefined') {
     this.updateDollarSource(context, update['$source'], update.timestamp);
   } else {
     console.error("No source in delta update:" + JSON.stringify(update));
   }
+
+  // second, update the values
   addValues(context, contextPath, update.source || update['$source'], update.timestamp, update.values);
 }
 
+/**
+ * Update the $source in the context.
+ *
+ * $source is a pointer to the sources field in the context.  See doc/data_model.html
+ *
+ * @param {Object} context
+ * @param {string} dollarSource a path directive pointing the real source
+ * @param {string} timestamp
+ */
 FullSignalK.prototype.updateDollarSource = function(context, dollarSource, timestamp) {
   const parts = dollarSource.split('.')
+  // descend into the sources element of the context, creating elements as needed
   parts.reduce((cursor, part) => {
     if(typeof cursor[part] === 'undefined') {
       return cursor[part] = {}
     }
     return cursor[part]
   }, this.sources)
+
+  // Uh, shouldn't something be done with the result of the reduce?  What if
+  // the pointed to value isn't found?
```

Comment:
You already documented what this does: _creating elements as needed_, using reduce cursor as a temp variable.

Reply:
I'm still confused.  All this does is create elements, it doesn't actually update anything.  Notice that context and timestamp aren't referenced in the body of the method.  (Which means my comment is wrong, it's descending into this.sources, not context.)

Reply:
Ok, so 
- it is handling this.sources and `context` is not needed nor used here, should be removed
- it is not updating timestamp, which it should be doing, so a bug - add a FIXME comment in this documentation PR?
- your comment about descending into sources of the context is not correct, so to get this PR merged please remove that